src/components/PopupConfirmation.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * TODO: Turn this into a generic confirmation dialogue,
+ *  instead of being specifically for card deletion
+ *  Instead of sending a function in at creation, 
+ *  send it in during open(). Then it can be used for
+ *  any sort of confirmation on the page.
+ */
+
+import Popup from './Popup.js';
+
+export default class PopupConfirmation extends Popup{
+  #buttonElement
+  #confirmAction
+  #card
+
+  constructor (popupSelector, confirmAction){
+    super(popupSelector);
+    this.#buttonElement = this.getPopupElement().querySelector('.modal__button')
+    this.#confirmAction = confirmAction;
+  }
+
+  #handleButtonClick = () => {
+    this.#confirmAction(this.#card)
+  }
+
+  /**
+   * Opens confirmation dialogue, and sets the target to card
+   * @param {Object} card 
+   */
+  open(card){
+    this.#card = card;
```

Comment:
Needs correction: `PopupWithConfirmation` is a general class. It can be used for any confirmation, not only card delete confirmation. This class should have a setter for `_handleFormSubmit`, not for card, so you can use a card in that handler in `index.js`.

```js
constructor(popupSelector) {
  super(popupSelector);
  this.#buttonElement = ..
}

setAction(action) { 
  this.#handleConfirm = action; 
} 

..
this.#buttonElement.addEventListener('mousedown', this.#handleConfirm)
```